content/player/blockchain-integration/buying-nfts.md

```diff
@@ -178,7 +178,7 @@ safeTransferFrom(
 );
 ```
 
-When the Bids contract receives the NFT, it will check in its `onERC721Received` function that the bid is not expired and the bidder has enough funds to pay for it. If everything is correct, the NFT will be transferred to the bidder and the previous owner, the address declared in the `from` parameter of the `safeTransferFrom` function, will be paid the amount declared in the Bid.
+When the Bids contract receives the NFT, it will check in its `onERC721Received` function that the bid is not expired and the bidder has enough funds to pay for it. If everything is correct, the NFT will be transferred to the bidder and the caller of the `safeTransferFrom` function will be paid the amount of MANA declared in the Bid.
```

Comment:
I would keep both. Something like

```
[...] previous owner, the address declared in the `from` parameter will be paid the amount declared in the Bid. Remember that `from` is the caller of the `safeTransferFrom` function.
```

Reply:
The from of the onERC721Received function is the caller, the from of the safeTransferFrom is the owner of the asset. If some address has been given approval for a token, it will be able to transfer the asset to the Bids contract and receive the MANA instead of the real owner.